server/services/offlinePayment.service.js

```diff
@@ -102,3 +103,112 @@ export const getAllOfflinePayments = async (user, { page = 1, limit = 10 }) => {
   const result = offlinePayments[0].data;
   return { pagination, result };
 };
+
+export const resolveOfflinePayment = async ({ offlinePaymentId, adminId }) => {
+  const offlinePayment = await getOfflinePaymentById(offlinePaymentId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+  if (!offlinePayment) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid offline payment');
+  }
+
+  const transaction = {
+    offerId: offlinePayment.offerId,
+    paymentSource: offlinePayment.type,
+    amount: offlinePayment.amount,
+    paidOn: offlinePayment.dateOfPayment,
+    additionalInfo: offlinePayment._id,
```

Comment:
Lets use something more explicit to totally differentiate offline payment info from others. Also, the offline payment  Id can still be retrieved by splitting the string and taking the second value.
```suggestion
    additionalInfo: `offlinePaymentId:${offlinePayment._id}`,
```